sympy/ntheory/factor_.py

```diff
@@ -2225,3 +2225,62 @@ def is_amicable(m, n):
         return False
     a, b = map(lambda i: divisor_sigma(i), (m, n))
     return a == b == (m + n)
+
+
+def eva(number, operation):
+    """Generates digits of a number."""
+    if operation == '*':
+        next_num = 1
+    else:
+        next_num = 0
+    while number > 0:
+        if operation == '*':
+            next_num = next_num * (number % 10)
+        else:
+            next_num = next_num + (number % 10)
+        number = int(number / 10)
+    return next_num
+
+
+def persistence(n,operation='*'):
+    """
+    The persistence of a number is the number of times one must apply
```

Comment:
This is discussed in a general way but applied to only addition and multiplication. Perhaps it would be better to have `dra=digital_root_add` and `drm=digital_root_mul`. Like `factorint` perhaps a visual flag could be used to request the iterative results vs the final result.

Reply:
Persistence is usually applied to addition and multiplication operation. I was not able to find any more type of persistence. What is the iterative result.

Reply:
@jksuom  please give me suggestions for this pr.

Reply:
Rather than having an operation flag I would just make two separate functions. I think @smichr is suggesting the same and that they should be called `drm` and `dra` for digital root multiply/add:
[https://en.wikipedia.org/wiki/Digital_root](https://en.wikipedia.org/wiki/Digital_root)

Reply:
Those functions should return ints rather than strings. There could be a separate function to return the sequence leading to the root.

Reply:
Okay, i will change it.